src/notifications/types.ts

```diff
@@ -0,0 +1,32 @@
+export interface ActionDefinition {
+    /** Type of the button the button, it can be go-to-url etc. */
+    type: string
+    /** Url of the button if the button is to open a new link */
+    url?: string
+    /** It is used for open the new tab or self tab {new-tab|self} */
+    context?: string
```

Comment:
Note you can use string literals to type things, which would be appropriate here. eg: `context?: 'new-tab' | 'self'`

You should also do it for the `type` field. BUT I think it's probably better to define the type in the `notifications/action-types` module (you'll have to declare the string literals twice because of language limitations, so it's better they're done in the same place to avoid errors later when we add more types). You could just do something like `export type ActionType = 'go-to-url' | 'toggle-storage-option'` from that module, then import it in here and use it on this line like `type: ActionType`.